features/step_definitions/signup_definitions_step.rb

```diff
@@ -0,0 +1,19 @@
+Given("I am on the landing page") do
+  visit root_path 
+end
+
+When("I click on the {string} button") do |string|
+  click_on "Sign up"
+end
+
+Then("I fill in {string} with {string}") do |name, user_name|
+  fill_in(name, with: "Bob")
+end
+
+When("I click {string}") do |string|
+  click_on "Create"
+end
+
+Then("I should be on the landing page") do
+  visit root_path
```

Comment:
Here you are making the assertion that you are on the landing page, when you write `visit root_path` you are redirecting the user to the landing page what you want to do here is to check if the user is on the landing page.
```
expect(current_path).to eq root_path
```